packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -305,14 +372,30 @@ export class Secret extends SecretBase {
    * @param attrs the attributes of the imported secret.
    */
   public static fromSecretAttributes(scope: Construct, id: string, attrs: SecretAttributes): ISecret {
-    class Import extends SecretBase {
-      public readonly encryptionKey = attrs.encryptionKey;
-      public readonly secretArn = attrs.secretArn;
-      public readonly secretName = parseSecretName(scope, attrs.secretArn);
-      protected readonly autoCreatePolicy = false;
+    let secretArn: string;
+    let secretArnIsPartial: boolean;
+
+    if (attrs.secretArn) {
+      if (attrs.secretCompleteArn || attrs.secretPartialArn) {
+        throw new Error('cannot use `secretArn` with `secretCompleteArn` or `secretPartialArn`');
+      }
+      secretArn = attrs.secretArn;
+      secretArnIsPartial = false;
+    } else {
+      if ((attrs.secretCompleteArn && attrs.secretPartialArn) ||
+          (!attrs.secretCompleteArn && !attrs.secretPartialArn)) {
+        throw new Error('must use only one of `secretCompleteArn` or `secretPartialArn`');
+      }
+      [secretArn, secretArnIsPartial] = attrs.secretCompleteArn ? [attrs.secretCompleteArn, false] : [attrs.secretPartialArn!, true];
```

Comment:
I wonder - if `secretCompleteArn` was passed, should we validate it with `arnIsComplete`, and throw if it doesn't match?

Reply:
And same in `fromSecretCompleteArn()`?

Reply:
Yes, I originally had that, then removed it for fear of false positives from `fromSecretArn`; however, that was before I added the `arnIsComplete` check within `fromSecretArn` itself. Check re-added.